### PR TITLE
DurationFormat: Add tests for durations with `style: "digital"`, `hoursDisplay: "auto"` and zero hours

### PR DESCRIPTION
**Pull request by sosukesuzuki**
The implementation of `Intl.DurationFormat` in WebKit (JavaScriptCore) has passed all test262 test cases, but we discovered two incompatibilities with the specification. This Pull Request adds tests addressing those issues.

The first one is described in [bug 285078](https://bugs.webkit.org/show_bug.cgi?id=285078). A separator is printed erroneously for an unprinted hours value:

```js
const df = new Intl.DurationFormat('en-US', {
  style: "digital",
  hoursDisplay: "auto",
});
df.format({hours: 0, minutes: 1, seconds: 2});
// expected: "01:02"
// actual  : ":, 01:02"
```

The second one is described in [bug 285212](https://bugs.webkit.org/show_bug.cgi?id=285212). A negative sign is not printed:

```js
const df = new Intl.DurationFormat('en-US', {
  style: "digital",
  hoursDisplay: "auto",
});
df.format({ hours: 0, minutes: -2, seconds: -3 });
// expected: "-02:03"
// actual  : "02:03"
```

Both issues occur when `style: "digital"`, `hoursDisplay: "auto"`, and the hours value is 0 or absent. For that reason, they are combined into a single test file.

### CODE REVIEW

File: test/intl402/DurationFormat/prototype/format/digital-style-with-hours-display-auto-with-zero-hour.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const df = new Intl.DurationFormat("en", {
+  style: "digital",
+  hoursDisplay: "auto",
+});
+
+const durations = [
+  // basic zero hours
+  [{ hours: 0, minutes: 0, seconds: 2 }, "00:02"],
+  [{ hours: 0, minutes: 1, seconds: 2 }, "01:02"],
+  [{ days: 1, hours: 0, minutes: 1, seconds: 2 }, "1 day, 01:02"],
```
Comment (sosukesuzuki) [inline test/intl402/DurationFormat/prototype/format/digital-style-with-hours-display-auto-with-zero-hour.js:21]: Chrome prints `"1 day:01:02"` for this, which is correct?

**Comment by sosukesuzuki** [comment]
esvu on CI fails but probably this PR isn't related

```
esvu ❯ version 1.2.16
JavaScriptCore ❯ Checking version...
esvu ✖ FetchError: request to https://webkitgtk.org/jsc-built-products/x86_64/release/LAST-IS failed, reason: certificate has expired
    at ClientRequest.<anonymous> (/home/circleci/test262/node_modules/node-fetch/lib/index.js:1501:11)
    at ClientRequest.emit (node:events:518:28)
    at emitErrorEvent (node:_http_client:101:11)
    at TLSSocket.socketErrorListener (node:_http_client:504:5)
    at TLSSocket.emit (node:events:518:28)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  type: 'system',
  errno: 'CERT_HAS_EXPIRED',
  code: 'CERT_HAS_EXPIRED'
}
esvu ✖ Some engines were not installed.
```

File: test/intl402/DurationFormat/prototype/format/digital-style-with-hours-display-auto-with-zero-hour.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  [{ days: 1,  minutes: 1, seconds: 2 }, "1 day, 01:02"],
+
+  // negative sign
+  [{ hours: 0, minutes: -1, seconds: -2 }, "-01:02"],
+  [{ hours: 0, minutes: -1, seconds: -2 }, "-01:02"],
+  [{ days: -1, hours: 0, minutes: -1, seconds: -2 }, "-1 day, 01:02"],
+];
+
+for (const [duration, expected] of durations) {
+  assert.sameValue(df.format(duration), expected, `Duration is ${duration}`);
```
Comment (anba) [inline test/intl402/DurationFormat/prototype/format/digital-style-with-hours-display-auto-with-zero-hour.js:35]:
```suggestion
  assert.sameValue(df.format(duration), expected, `Duration is ${JSON.stringify(duration)}`);
```

Otherwise `${duration}` will evaluate to the string `"[object Object]"`.

File: test/intl402/DurationFormat/prototype/format/digital-style-with-hours-display-auto-with-zero-hour.js (hunk 1 above)
Comment (anba) [reply test/intl402/DurationFormat/prototype/format/digital-style-with-hours-display-auto-with-zero-hour.js:21]: That sounds like a V8 bug. 

In [FormatNumericUnits](https://tc39.es/proposal-intl-duration-format/#sec-formatnumericunits), steps 10-11 will set `hoursFormatted` to `false`, because the hour value is zero and the hours display is not `"always"`. Step 17 passes `hoursFormatted` to [FormatNumericMinutes](https://tc39.es/proposal-intl-duration-format/#sec-formatnumericminutes). FormatNumericMinutes, step 2 only appends the separator if `hoursDisplayed` is `false`. And the value of `hoursDisplayed` is `hoursFormatted` from FormatNumericUnits. That means no separator should be appended for this case.

**Review by anba — APPROVED** [review]

**Review by Ms2ger — APPROVED** [review]
Happy to defer to anba here

**Comment by sosukesuzuki** [comment]
@anba Thank you for review!

**Comment by sosukesuzuki** [comment]
@Ms2ger Anba's suggestions have been applied, probably we can merge this
